apps/designer/app/shared/remix/trpc-remix-proxy.ts

```diff
@@ -16,33 +17,46 @@ export const createTrpcRemixProxy = <Router extends AnyRouter>(
         useMutation: () => {
           submit: (input: inferRouterInputs<Router>[Procedure]) => void;
           data?: inferRouterOutputs<Router>[Procedure];
+          state: "idle" | "loading" | "submitting";
         };
       }
     : {
         useQuery: () => {
           load: (input: inferRouterInputs<Router>[Procedure]) => void;
           data?: inferRouterOutputs<Router>[Procedure];
+          state: "idle" | "loading" | "submitting";
         };
       };
 } =>
   createRecursiveProxy((options) => {
     const path = [...options.path];
     const hook = path.pop();
+    const { data, submit, state } = useFetcher();
 
     if (hook !== "useMutation" && hook !== "useQuery") {
       throw new Error(`Invalid hook ${hook}`);
     }
 
-    const fetcher = useFetcher();
-
     const method = path.join(".");
 
-    const submit = (input: never) => {
-      return fetcher.submit(input, {
-        method: hook === "useMutation" ? "post" : "get",
-        action: getPath(method),
-      });
-    };
+    const submitOrLoad = useCallback(
```

Comment:
I was just discussing with Chance from Remix team that submit is designed to be used for FormData, we should call it send() instead

Reply:
submitOrLoad is also not very nice, I think  you could just have a separate function for .load()

Reply:
function is same renamed to `remixSubmit` its internal function to call remix submit